tests/ha/firewall_disable.pm

```diff
@@ -14,14 +14,26 @@ use base 'opensusebasetest';
 use strict;
 use testapi;
 use hacluster;
+use version_utils 'is_sle';
+
+sub stop_and_disable_firewall {
+    my $firewall = shift;
+
+    if (is_package_installed "$firewall") {
+        assert_script_run "systemctl -q is-active $firewall && systemctl disable $firewall; systemctl stop $firewall";
+        return 0;
+    }
+
+    return 1;
+}
 
 sub run {
     my ($self) = @_;
-    my $firewall = $self->firewall;
 
     # Deactivate firewall if needed
-    if (is_package_installed "$firewall") {
-        assert_script_run "systemctl -q is-active $firewall && systemctl disable $firewall; systemctl stop $firewall";
+    if (stop_and_disable_firewall $self->firewall) {
+        # We can be in a specific case when we are on a SLE>=15 upgraded from a SLE<=15 image
```

Comment:
Wouldn't this be covered by L492 from `lib/opensusebasetest.pm`?

Reply:
I need to check why but it wasn't working, due to the fact the during an upgrade from sle12sp3 to sle15 VERSION is set to 15 but firewall in the upgraded OS is still the old one (for compatibility reason). I just need to retest the upgrade process to check why I do like this.

Reply:
Checked some more, and while the regular expression in  L492 of `lib/opensusebasetest.pm` does catch this, the `is_upgrade()` call on the same line returns false. I guess you're upgrading a SLES-12 SP3 qcow2 image in a test and publishing the HDD, and then performing the HA tests in a separate run.

So, IMO we have 3 or 4 options to handle this:

- Remove the call to `is_upgrade()` from L492 in `lib/opensusebasetest.pm`. No idea what it'll break.
- Add a new ENV variable and set it when using an upgraded system qcow2 from another test. Something like UPGRADED_SYSTEM or similar. Then add a check for that variable either in `firewall()` in `lib/opensusebasetest.pm` or in `is_upgrade()` in `lib/version_utils.pm`. No idea what it'll break if the change is added to `is_upgrade()`, but it should be fairly harmless in `firewall()`.
- Attempt to disable both `SuSEfirewall2` and `firewalld` in `tests/ha/firewall_disable.pm`. This should be pretty harmless, but the test will always be attempting to disable something that's not actually enabled because it's an either-or situation. This is not so different what you're currently doing, which is code to disable `SuSEfirewall2` when it fails to disable `firewalld`.

I guess we should ping @asmorodskyi, @Soulofdestiny and @okurz regargind this as well.

Sorry for the lengthy reply.

Reply:
Removing the call to `is_upgrade()` in L492 of `lib/opensusebasetest.pm` is not actually a valid option. Updated the comment to reflect that.

Perhaps something can be done with the OS version during the test and the version from the qcow2 image though.

Reply:
Yes, the existing use of HDD_VERSION or something similar might help you there. You can consult with the experts from QA SLE migration here

Reply:
According to the `firewall` function code in `lib/opensusebasetest.pm`, it seems that if I set the *old* version number in the `HDD_1`  variable, `firewall` should return the correct version.

In my test, no version number was set in `HDD_1`, so I think it's better to use the existing solution and let my script in his previous state.